CMakeLists.txt

```diff
@@ -80,9 +81,13 @@ string(APPEND CMAKE_CXX_FLAGS " -Wbuiltin-macro-redefined")
 string(APPEND CMAKE_CXX_FLAGS " -pedantic")
 string(APPEND CMAKE_CXX_FLAGS " -Werror")
 string(APPEND CMAKE_CXX_FLAGS " -fno-omit-frame-pointer")
-if(LEAKCHECK AND THREADCHECK)
+if((LEAKCHECK AND THREADCHECK) OR (ADDRESSCHECK AND THREADCHECK))
     message(FATAL_ERROR "Cannot have both Thread and Address/Leak Sanitizers Enabled")
 endif()
+if(ADDRESSCHECK)
+    string(APPEND CMAKE_CXX_FLAGS " -fsanitize=address")
+    message("-- Address Sanitizers Enabled")
+endif()
```

Comment:
Is this change intentionally included in the PR?

Reply:
Unintentionally included :) will revert